go/libraries/doltcore/sqle/dolt_map_iter.go

```diff
@@ -0,0 +1,137 @@
+package sqle
+
+import (
+	"context"
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+type KVToSqlRowConverter struct {
+	tagToSqlColIdx map[uint64]int
+	cols           []schema.Column
+	rowSize        int
+}
+
+func NewKVToSqlRowConverterForCols(cols []schema.Column) *KVToSqlRowConverter {
+	tagToSqlColIdx := make(map[uint64]int)
+	for i, col := range cols {
+		tagToSqlColIdx[col.Tag] = i
+	}
+
+	return &KVToSqlRowConverter{
+		tagToSqlColIdx: tagToSqlColIdx,
+		cols:           cols,
+		rowSize:        len(cols),
+	}
+}
+
+func (conv *KVToSqlRowConverter) ConvertKVToSqlRow(k, v types.Value) (sql.Row, error) {
+	keyTup := k.(types.Tuple)
+	var valTup types.Tuple
+	if !types.IsNull(v) {
+		valTup = v.(types.Tuple)
+	}
+
+	cols := make([]interface{}, conv.rowSize)
+	filled, err := conv.processTuple(cols, 0, keyTup)
+	if err != nil {
+		return nil, err
+	}
+
+	if !valTup.Empty() {
+		filled, err = conv.processTuple(cols, filled, valTup)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return cols, err
+}
+
+func (conv *KVToSqlRowConverter) processTuple(cols []interface{}, filled int, tup types.Tuple) (int, error) {
+	tupItr, err := tup.Iterator()
+
+	if err != nil {
+		return 0, err
+	}
+
+	for filled < len(conv.tagToSqlColIdx) {
```

Comment:
conv.rowSize instead of len()

Reply:
nah they are not equivalent.  will add some needed comments.